core/engine/src/value/conversions/serde_json.rs

```diff
@@ -113,6 +113,15 @@ impl JsValue {
     ///
     /// Panics if the `JsValue` is `Undefined`.
     pub fn to_json(&self, context: &mut Context) -> JsResult<Value> {
+        let mut stack = vec![];
+        self.to_json_inner(context, &mut stack)
+    }
+
+    fn to_json_inner(
+        &self,
+        context: &mut Context,
+        stack: &mut Vec<JsObject<dyn NativeObject>>,
```

Comment:
```suggestion
        stack: &mut Vec<JsObject>,
```
Nitpick: I don't think you manually have to specify the inner type since it's default.